packages/ember-testing/lib/helpers.js

```diff
@@ -89,9 +89,13 @@ function check(app, selector, context) {
   Ember.assert('To check \'' + selector +
       '\', the input must be a checkbox', type === 'checkbox');
 
-  if (!$el.prop('checked')) {
-    app.testHelpers.click(selector, context);
-  }
+  Ember.assert('Expected \'' + selector +
+      '\'to be unchecked before click, but was checked', !$el.prop('checked'));
+
+  app.testHelpers.click(selector, context);
+
+  Ember.assert('Expected \'' + selector +
```

Comment:
Since `click` is async, I believe that this assertion will also need to be async.